### PR TITLE
Quo2 Wallet: Network Link

### LINKED ISSUE #16754: Quo2 Wallet: Network Link component
Implement this component in the quo2 component library as per the design spec.

Reuse Quo2 components wherever applicable.

Designs
https://www.figma.com/file/WQZcp6S0EnzxdTL4taoKDv/Design-System-for-Mobile?type=design&node-id=22568-205044&mode=design&t=Yui6CQ6Hh2CplG7a-0

Acceptance Criteria:

component spec tests - https://github.com/status-im/status-mobile/blob/develop/doc/component-tests-overview.md
quo2 preview screen
export from quo2/core.cljs
following best practices - https://github.com/status-im/status-mobile/blob/develop/doc/new-guidelines.md

Note
Please align with design team before starting this

### PR DESCRIPTION
**Pull request by OmarBasem**
fixes: https://github.com/status-im/status-mobile/issues/16754

This PR implements Network Link component.

[Designs](https://www.figma.com/file/WQZcp6S0EnzxdTL4taoKDv/Design-System-for-Mobile?type=design&node-id=22568-205044&mode=design&t=gziSonRgaeAt4fTW-0)

Preview:

https://github.com/status-im/status-mobile/assets/29354102/0406dbf9-2ec0-4dad-b07f-1b2176bd29a5

### CODE REVIEW

File: src/status_im2/contexts/quo_preview/wallet/network_link.cljs [hunk 1]
```diff
@@ -0,0 +1,33 @@
+(ns status-im2.contexts.quo-preview.wallet.network-link
+  (:require
+    [quo2.core :as quo]
+    [react-native.core :as rn]
+    [reagent.core :as reagent]
+    [status-im2.contexts.quo-preview.preview :as preview]))
+
+
```
Comment (smohamedjavid) [inline src/status_im2/contexts/quo_preview/wallet/network_link.cljs:15]: Extra line

**Review by smohamedjavid — APPROVED** [review]
LGTM 👍

File: src/quo2/components/wallet/network_link/view.cljs [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+(def height
+  {:linear 10
+   :1x     66
+   :2x     122})
+
+(defn- view-internal
+  [{:keys [shape theme preview?]}]
+  ;; Updating image height on the fly (in preview screen) on Android does not work with fast-image
+  ;; so we are using rn/image for preview screen
+  (let [image-component (if (and platform/android? preview?) rn/image fast-image/fast-image)
```
Comment (J-Son89) [inline src/quo2/components/wallet/network_link/view.cljs]: hmm I'm not sure if this is supposed to be an image - is it more of an svg type component? 🤔 
did you check the expected behaviour with the designers ?

Also I think the colors are not quite fixed but more of a linear gradient that depends on some configurations

**Review by J-Son89 — CHANGES_REQUESTED** [review]
check my comment about the approach - I think it should be svg type component? 🤔

**Review by vkjr — CHANGES_REQUESTED** [review]
Agree with @J-Son89, that clarifications from designer team are required

**Comment by OmarBasem** [comment]
@J-Son89 @vkjr fixed it 👍

**Review by J-Son89 — APPROVED** [review]
looks great, nice @OmarBasem! :)

**Comment by OmarBasem** [comment]
@Francesca-G design review please :)

**Review by Francesca-G — APPROVED** [review]
Looks good ✨

Adding required follow up due to minor details:
- the x2 style is shorter than design (see screenshot)
- the circles look slightly smaller

Screenshot 2023-08-22 alle 09 33 55

Design on the left, implementation on the right

**Comment by OmarBasem** [comment]
@vkjr re-review please

**Review by vkjr — APPROVED** [review]

**Comment by vkjr** [comment]
@OmarBasem, thanks!

**Comment by churik** [comment]
@OmarBasem  
 did you create an issue follow-up based on the design review (or maybe fixed this in Pr and didn't mention)?

Based on https://github.com/status-im/status-mobile/blob/develop/doc/pipeline_process.md > Design review, all follow-up issues have to be created when PR is merged and linked to the original PR.

**Comment by OmarBasem** [comment]
@churik yes there is a follow up https://github.com/status-im/status-mobile/issues/17088, and linked to the original PR


Screenshot 2023-09-25 at 07 50 39

**Comment by churik** [comment]
Oh, sorry, my bad!

**Pull request merged**